integration/kafka/src/main/java/alluxio/kafka/connect/AlluxioTopicPartitionWriter.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.kafka.connect;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.exception.AlluxioException;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.kafka.connect.format.AlluxioFormat;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Queue;
+import java.util.UUID;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * This class writes kafka's specific topic partition data to alluxio filesystem.
+ */
+public class AlluxioTopicPartitionWriter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AlluxioSinkTask.class);
+  private TopicPartition mTopicPartition;
+  private long mOffset;
+  private FileOutStream mFileOutStream;
+  private FileSystem mFs;
+  private long mRecordNum;
+  private String mFileTempPath;
+  private String mTopicPartitionPath;
+  private String mTmpTopicPartitionPath;
+  private AlluxioSinkConnectorConfig mConfig;
+  private SinkTaskContext mContext;
+  private AlluxioFormat mFormat;
+  private long mLastRotationTime;
+  private long mRotationRecordNum;
+  private long mRotationTimeInterval;
+  private Queue<SinkRecord> mRecordQueue = new LinkedBlockingQueue<>();
```

Comment:
Is the writer used by multiple threads? Seems only used by one task. Probably used a LinkedList instead?